examples/platform/asset_groups.py

```diff
@@ -0,0 +1,191 @@
+    for ag in all_asset_groups:
+        print("asset group name: {}, member count: {}".format(ag.name, ag.member_count))
+
+    # Create an asset group.  The only mandatory field is the Asset Group Name.
+    # It can be created without a policy, which enables the use of group for visibility of specific assets
+    # It can be created without a query, which enables manual assignment of assets to the group later
+    new_asset_group_name = "My Example Asset Group"
+    new_asset_group = AssetGroup.create_group(api, new_asset_group_name, "Demonstrating the SDK")
+    print(new_asset_group)
+
+    # Add a query.  All assets that match this criteria will be dynamically added to the group
+    new_asset_group.query = "os.equals:WINDOWS"
+    # Assign a policy.  All assets in the group may have this policy applied.  If an asset is in more than one group,
+    # policy ranking determines which is the effective policy.
+    bottom_rank_policy = None
+    for p in api.select(Policy).all():
+        if bottom_rank_policy is None or p.position > bottom_rank_policy.position:
+            bottom_rank_policy = p
+    new_asset_group.policy = bottom_rank_policy.id
+    new_asset_group.save()
+    print("\n\n new_asset_group {}".format(new_asset_group))
+    # Clean up after ourselves and delete the asset group
+    new_asset_group.delete()
+
+    # An asset group can also be created with a query and / or a policy included
+    print("\n\n Second asset group with policy and query")
+    second_name = "Second demo group"
+    second_asset_group = AssetGroup.create_group(api, second_name, "Second group description",
+                                                 query="os.equals:MAC", policy_id=bottom_rank_policy.id)
+    second_asset_group.refresh()
+    while second_asset_group.status != "OK":
+        print("waiting")
+        time.sleep(10)
+        second_asset_group.refresh()
+
+    # Asset groups can be searched
+    search_asset_group_query = api.select(AssetGroup).add_criteria("name", second_name).sort_by("name", "ASC")
+    for ag in search_asset_group_query:
+        print("\n\nAsset group name = {}. It has {} members".format(ag.name, ag.member_count))
+        print("Policy assigned to the Asset Group is Name: {}, Id: {}".format(ag.policy_name, ag.policy_id))
+        # These are the assets that are now part of the dynamic asset group
+        for d in ag.list_members():
+            print("Device Name: {}, Id: {}".format(d.name, d.id))
+            if d.policy_id == bottom_rank_policy.id:
+                print("The effective policy is from the asset group")
+            else:
+                print("This asset group does not determine the effective policy The effective policy is {} - {}"
+                      .format(d.policy_id, d.policy_name))
+
+    # Clean up after ourselves and delete the asset group
+    second_asset_group.delete()
+
+    # Assets can be assigned manually to an asset group rather than with a query
```

Comment:
Note an Asset group can have both manual and dynamic members

Reply:
Changed to use second asset group again and add the device.